### PR TITLE
[JWT Grant] Multi-Environment Authentication with login to one environment

### PR DESCRIPTION
**Pull request by RenukaFernando**
#### Changes in the backend
- Create DCR app with JWT grant type.
- Add `assertion` form param to the token endpoint.
- Implement `WSO2ISKeyManagerImpl` to support JWT from WSO2-IS.
- Testcase for the `AuthenticatorService` and `WSO2ISKeyManagerImpl`
- Change environment configuration to the new namespace `wso2.carbon.apimgt.environment`.
sample configuration
```yaml
  # APIM Environment Configuration Parameters
wso2.carbon.apimgt.environments:
    # current environment's label from the list of environments
  environmentLabel: Production
    # UI Service URL
  allowedHosts:
  - 10.100.4.247:9292
  - 10.100.4.177:9292
  - localhost:9292

    # Multi-Environment Overview Configurations
  multiEnvironmentOverview:
      # Multi-Environment Overview feature enabled or not
    enabled: true
```
#### Changes in the frontend
- Remove `idToken` from local-storage.
- Authenticate the user to all other environments.

Fix https://github.com/wso2/product-apim/issues/2629
Fix https://github.com/wso2/product-apim/issues/2694

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/configuration/models/EnvironmentConfigurations.java [hunk 1]
```diff
@@ -1,3 +1,21 @@
+/*
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/configuration/models/EnvironmentConfigurations.java]: Shall we update the license header?

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/internal/ServiceReferenceHolder.java [hunk 2]
```diff
@@ -59,20 +61,43 @@ public void setConfigProvider(ConfigProvider configProvider) {
...
+     * Gives the EnvironmentConfigurations explicitly set in the deployment yaml or the default configurations
+     *
+     * @return EnvironmentConfigurations
+     */
+    public EnvironmentConfigurations getEnvironmentConfigurations() {
+        try {
+            if (configProvider != null) {
+                environmentConfigurations = configProvider.getConfigurationObject(EnvironmentConfigurations.class);
+            } else {
+                log.error("Configuration provider is null");
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/internal/ServiceReferenceHolder.java]: Please link the github issue for the issue in this code

**Comment by RenukaFernando** [comment]
fixed UI customizations with https://github.com/wso2/carbon-apimgt/pull/5010/commits/a64caf28842a0383073251b1c6efa517fa9deef7

**Comment by Mariangela** [comment]
SSO does not work with multi-environment authentication.
https://github.com/wso2/product-apim/issues/2736

**Comment by Mariangela** [comment]
Wiki Documentation: https://docs.wso2.com/display/AM300/Maintaining+Multiple+Environments

**Pull request merged**